src/qibo/transpiler/unroller.py

```diff
@@ -240,6 +249,13 @@ def _translate_two_qubit_gates(gate: gates.Gate, native_gates: NativeGates):
                 iswap_decomposed.append(g_translated)
         return iswap_decomposed
 
+    # For testing purposes
+    # No CZ, iSWAP gates in the native gate set
+    # Decompose CNOT, CZ, SWAP gates into CNOT gates
+    if native_gates & NativeGates.CNOT:
+        return cnot_dec_temp(gate)
```

Comment:
You can make this decomposition work for every gate by first decomposing into CZ and then decomposing CZ into CNOT. This is how it works for iSWAP decomposition (line 238-250).

Reply:
Yes, but it adds two H gates next to the CNOT and it requires optimization of 1q gates. Since Paul will use simple circuits for testing, I only added basic decompositions. As @alecandido mentioned, it would be better to entirely review the unroller.

Reply:
Yes, but it adds two H gates next to the CNOT and it requires optimization (fusion) of 1q gates. Since Paul will use simple circuits for testing, I only added basic decompositions. As @alecandido mentioned, it would be better to entirely review the unroller.

Reply:
It should be a realy easy check to remove two successive H gates, however if you need it only for simple applications it is fine